src/Core/Vault/Authorization/Permissions/NormalCipherPermissions.cs

```diff
@@ -0,0 +1,33 @@
+﻿#nullable enable
+using Bit.Core.Entities;
+using Bit.Core.Models.Data.Organizations;
+using Bit.Core.Vault.Models.Data;
+
+namespace Bit.Core.Vault.Authorization.Permissions;
+
+public class NormalCipherPermissions
+{
+    public static bool CanDelete(User user, CipherDetails cipherDetails, OrganizationAbility? organizationAbility)
+    {
+        if (user.Id == cipherDetails.UserId)
+        {
+            return true;
+        }
+
+        if (organizationAbility == null)
+        {
+            throw new Exception("Cipher needs to belong to a user or an organization.");
+        }
```

Comment:
This error message doesn't describe the problem: the problem isn't the cipher, it's that no OrgAbility has been provided by the caller. You could rephrase the error message, e.g. "OrganizationAbility must not be null", or throw an `ArgumentNullException`, or if you want

Reply:
Thinking about this more, I believe we're trying to solve two problems here.
- The cipher does not have an owner:
    1. The `cipherDetails` does not have a `userId` or `organizationId`
- The cipher does not belong to the `organizationAbility`:
    1. The client may pass in the wrong organization.
    2. The client may pass `null` for `organizationAbility`.
  
I updated the code and added tests to cover both cases. Let me know what you think.

Reply:
I also noticed that it would probably be confusing to throw an exception for `organizationAbility` being null since it's a nullable parameter.